Extension/src/LanguageServer/client.ts

```diff
@@ -405,6 +404,7 @@ class DefaultClient implements Client {
             telemetry.logLanguageServerEvent("missingLanguageServerBinary");
             throw String('Missing binary at ' + serverModule);
         }
+        this.rootFolder = workspaceFolder;
```

Comment:
wait.. If you are already passing it in, why does where you set it matter? Why is it failing in the first place? I assume that since you pass it in, instead of using this.rootFolder, that you should be operating on workspaceFolder instead.

I feel like someone from your team needs to look at this.

Reply:
Code needs to references the this.rootFolder specifically, i.e. the lifetime of the workspaceFolder variable is only that specific function, so assigning to this.rootFolder extends the life of the variable when the method returns.

Reply:
I added the ${workspaceStorage} case too.